openmls/src/group/core_group/apply_proposals.rs

```diff
@@ -57,6 +57,35 @@ impl CoreGroup {
         let mut self_removed = false;
         let mut external_init_secret_option = None;
 
+        // Process external init proposals. We do this before the removes, so we
+        // know that removing "ourselves" (i.e. removing the group member in the
+        // same leaf as we are in) is valid in this case. We only care about the
+        // first proposal and ignore all others.
+        if let Some(queued_proposal) = proposal_queue
+            .filtered_by_type(ProposalType::ExternalInit)
```

Comment:
Wouldn't this be a good place to plug in the validation check that there is at most one external init per commit?

Reply:
Probably. I'm currently preparing a follow up that deals with validation (#680), but I haven't gone past signature verification yet. The validation flow of `CoreGroup` is not exactly straight-forward for external messages.